### ISSUE TITLE
Incorrectly reading/writting integer

### ISSUE THREAD

**Issue by FrogTheFrog** [tirish/steam-shortcut-editor#9]
Hello,

once again with and error. However, I have solution! The problem is that you read integer with `charCodeAt` which uses UTF-16. Now, the problem is that parsed string has no encoding, it's just 1 byte data "stream". `charCodeAt` will parse any byte above 128 as it should according to UTF-16. That's not what we want. Use the function below to read data properly:

```
function parseIntValue(input){
    var value = Buffer.from(input.str.substring(input.i, input.i + 4)).readInt32LE();

    input.i += 4;
    return value;
}
```

**Edit**:
Something is still broken. And I might be wrong with above statement.

**Comment by FrogTheFrog** [comment]
It causes random errors in unit test. Going away, so can't tell if it is my fault. May need to change something else.

**Comment by FrogTheFrog** [comment]
Attaching file, that has "LastPlayTime" changed otherwise.

[shortcuts5.zip](https://github.com/tirish/steam-shortcut-editor/files/979776/shortcuts5.zip)

**Comment by FrogTheFrog** [comment]
Ok, so increasing index by 4 fails, because it increases by characters. My file's `LastPlayTime` has 4 bytes, but 3 characters, thus corrupting any further reading.

**Comment by FrogTheFrog** [comment]
Honestly, I'm kinda lost at what is happening.
Attaching some files.
1. Is created by this parser.
2. Is edited by steam.
3. Is parsed by this parser and then stringyfied. `LastPlayTime` is different? Because Steam fails to read it. Is it utf-8 that messes it up?

[what.zip](https://github.com/tirish/steam-shortcut-editor/files/980160/what.zip)

**Comment by FrogTheFrog** [comment]
There is just no way around this. I'm going to rewrite this parser with buffer object instead of string. If it works, I'll pull request.

**Comment by tirish** [comment]
Hey @FrogTheFrog thanks for flagging, sorry about this. I have rewritten the parser to use a buffer instead.  I will v3 this when I am done and link here.

**Comment by FrogTheFrog** [comment]
I also just did that :)

**Comment by FrogTheFrog** [comment]
@tirish It's [here](https://github.com/FrogTheFrog/steam-shortcut-editor) in case you need it for something.

**Comment by tirish** [comment]
@FrogTheFrog looks pretty similar to my changes here: https://github.com/tirish/steam-shortcut-editor/issues/10 👍 

Still need to rip out the string read/write. Also I see you decided to treat empty strings as `undefined` rather than empty string. That is consistent with what this has been doing, but I now think empty string might make more sense.

Anyways, thanks for the help on this. I am going to push v3 soon-ish with a rather different looking API (since `stringify` seems a bit misleading for returning a `Buffer`)

**Comment by FrogTheFrog** [comment]
@tirish I'm going to sleep now. I hope that "soon-ish" period of time will be over after ~9 hours :) .

**Comment by tirish** [comment]
@FrogTheFrog version `3.0.0` has been pushed. See README for changes. 

TLDR: the API is now..
- `shortcut.parseFile`
- `shortcut.parseBuffer`
- `shortcut.writeFile`
- `shortcut.writeBuffer`

Let me know if you continue to have issues.
